src/mono/mono/mini/interp/transform.c

```diff
@@ -7576,6 +7576,12 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 			case CEE_UNUSED57: ves_abort(); break;
 #endif
 			case CEE_ENDFILTER:
+				td->sp--;
```

Comment:
Why are we decreasing the stack pointer once more ?

Reply:
I followed the same approach as in mono jit where the error is thrown:
https://github.com/dotnet/runtime/blob/a83c661bb8061b65404fb154b328ff1d3993d9ec/src/mono/mono/mini/method-to-ir.c#L11786-L11788

I understood that on the top of stack is a result of filter and it should be the only value on the stack in new BB. So, if there is something else, just throw.

Reply:
You are modifying the value of the stack pointer, which breaks following code, when you should only be checking what it is at that location.

Reply:
Just to check if I got it right, `sp - 1` should be equal to `stack_start` and `sp->type` should be `int32`?